apps/native-component-list/src/screens/BarCodeScannerScreen.tsx

```diff
@@ -39,17 +37,24 @@ function reducer(state: State, action: Partial<State>): State {
 }
 
 export default function BarcodeScannerScreen() {
-  const [isPermissionsGranted] = usePermissions(BarCodeScanner.requestPermissionsAsync);
+  const [permission, requestPermission] = usePermissions();
```

Comment:
Can also make it auto-request when loading the screen, with the button as fallback (if the user canceled that). For that, we only have to add `usePermissions({ request: true })`.

<img width="503" alt="Screenshot 2021-12-07 at 16 36 46" src="https://user-images.githubusercontent.com/1203991/145109860-9c5450f0-11c9-4007-80f7-202e34a4b103.png">

Reply:
![neat](https://i.imgur.com/IJg51oE.gif)